json-endpoints/class.wpcom-json-api-get-site-endpoint.php

```diff
@@ -309,11 +309,11 @@ protected function render_response_key( $key, &$response, $is_user_logged_in ) {
 			case 'user_can_manage' :
 				$response[ $key ] = $this->site->user_can_manage();
 			case 'is_private' :
-				$response[ $key ] = $this->site->is_private();
+				$response[ $key ] = wpcom_is_coming_soon() ? is_private_blog() : $this->site->is_private();
```

Comment:
Shouldn't this change happens in the `$this->site->private()` function? As part of the SAL, it allows for different conditions on WP.com vs Jetpack?

Reply:
Or a new SAL function `$this->site->is_coming_soon()` that is added to Jetpack and WP.com that conditionally return what is expected.